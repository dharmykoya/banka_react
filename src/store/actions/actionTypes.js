export const AUTH_START = 'AUTH_START';
export const AUTH_SUCCESS = 'AUTH_SUCCESS';
export const AUTH_FAIL = 'AUTH_FAIL';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export const USER_ACCOUNT = 'USER_ACCOUNT';
export const USER_ACCOUNT_SUCCESS = 'USER_ACCOUNT_SUCCESS';
export const USER_ACCOUNT_FETCH_STARTED = 'USER_ACCOUNT_FETCH_STARTED';

export const TRANSACTION_HISTORY_START = 'TRANSACTION_HISTORY_START';
export const TRANSACTION_HISTORY_SUCCESS = 'TRANSACTION_HISTORY_SUCCESS';

