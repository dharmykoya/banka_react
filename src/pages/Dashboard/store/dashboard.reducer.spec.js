import nock from 'nock';

import * as actionTypes from '../../../store/actions/actionTypes';
import dashboardAction from './dashboard.action';

const { userAccountFetchStarted } = dashboardAction;

describe('auth actions', () => {
  afterEach(() => {
    nock.cleanAll();
  });
  it('should create an action to start dashboard fetching', () => {
    const expectedAction = {
      type: actionTypes.USER_ACCOUNT_FETCH_STARTED
    };
    expect(userAccountFetchStarted()).toEqual(expectedAction);
  });
});
