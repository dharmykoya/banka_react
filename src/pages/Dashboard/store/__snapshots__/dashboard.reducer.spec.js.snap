// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth actions should create an action that fetches a user details using the userId 1`] = `
Array [
  Object {
    "type": "USER_ACCOUNT_FETCH_STARTED",
  },
  Object {
    "type": "USER_ACCOUNT_SUCCESS",
    "userAccount": Object {
      "0": Object {
        "account_number": 2000000003,
        "balance": "170002.00",
        "created_on": "2019-04-28T11:49:09.165Z",
        "id": 4,
        "owner": 5,
        "status": "active",
        "type": "savings",
        "updated_at": "2019-04-28T11:49:09.165Z",
      },
      "user": Object {
        "email": "issacola57@gmail.com",
        "firstName": "isaac",
        "id": 5,
        "isAdmin": false,
        "lastName": "olayisade",
        "type": "client",
      },
    },
  },
]
`;

exports[`auth actions should return error when trying to fetch a request 1`] = `
Array [
  Object {
    "type": "USER_ACCOUNT_FETCH_STARTED",
  },
  Object {
    "type": "USER_ACCOUNT_SUCCESS",
    "userAccount": Object {
      "0": Object {
        "account_number": 2000000003,
        "balance": "170002.00",
        "created_on": "2019-04-28T11:49:09.165Z",
        "id": 4,
        "owner": 5,
        "status": "active",
        "type": "savings",
        "updated_at": "2019-04-28T11:49:09.165Z",
      },
      "user": Object {
        "email": "issacola57@gmail.com",
        "firstName": "isaac",
        "id": 5,
        "isAdmin": false,
        "lastName": "olayisade",
        "type": "client",
      },
    },
  },
  Object {
    "type": "USER_ACCOUNT_FETCH_STARTED",
  },
]
`;
